File: CS/GrowthWare.Framework.BusinessData/DataAccessLayer/Interfaces/IDGroups.cs
using GrowthWare.Framework.BusinessData.DataAccessLayer.Interfaces.Base;
using GrowthWare.Framework.Model.Enumerations;
using GrowthWare.Framework.Model.Profiles;
using System;
using System.Data;

namespace GrowthWare.Framework.BusinessData.DataAccessLayer.Interfaces
{
    interface IDGroups : IDDBInteraction
    {
        /// <summary>
        /// Gets a subset of information from the database 
        /// </summary>
        /// <param name="searchCriteria"></param>
        /// <returns></returns>
        DataTable Search(MSearchCriteria searchCriteria);

        /// <summary>
        /// Sets or gets the SecurityEntitySeqID
        /// </summary>
        int SecurityEntitySeqID { get; set; }

        /// <summary>
        /// GroupProfile
        /// </summary>
        MGroupProfile Profile { get; set; }

        /// <summary>
        /// GroupRoles
        /// </summary>
        MGroupRoles GroupRolesProfile { get; set; }

        /// <summary>
        /// Returns a DataTable of Group roles
        /// </summary>
        /// <returns>DataTable</returns>
        DataTable GetGroupRoles();

        /// <summary>
        /// Updates the Groups roles
        /// </summary>
        /// <returns>bool</returns>
        bool UpdateGroupRoles();

        /// <summary>
        /// Get's all of the groups for a given Security Entity
        /// </summary>
        /// <returns>DataTable</returns>
        DataTable GetGroupsBySecurityEntity();

        /// <summary>
        /// Adds a group to a Security Entity
        /// </summary>
        /// <returns>int</returns>
        void AddGroup();

        /// <summary>
        /// Returns a data row necessary to populate MGroupProfile
        /// </summary>
        /// <returns>DataRow</returns>
        DataRow GetProfileData();

        /// <summary>
        /// Deletes a group in a given Security Entity
        /// </summary>
        /// <returns>bool</returns>
        bool DeleteGroup();

        /// <summary>
        /// Saves this instance.
        /// </summary>
        void Save();
    }
}

File: VB/GrowthWare.Framework.BusinessData/DataAccessLayer/Interfaces/IDGroups.vb
Imports GrowthWare.Framework.BusinessData.DataAccessLayer.Interfaces.Base
Imports GrowthWare.Framework.Model.Profiles

Namespace DataAccessLayer.Interfaces
    Public Interface IDGroups
        Inherits IDDBInteraction

        ''' <summary>
        ''' Searches the specified search critera.
        ''' </summary>
        ''' <param name="searchCriteria">The search critera.</param>
        ''' <returns>DataTable.</returns>
        Function Search(ByVal searchCriteria As MSearchCriteria) As DataTable

        ''' <summary>
        ''' Gets or sets the security entity seq ID.
        ''' </summary>
        ''' <value>The security entity seq ID.</value>
        Property SecurityEntitySeqId As Integer

        ''' <summary>
        ''' Deletes the group.
        ''' </summary>
        Sub DeleteGroup()

        ''' <summary>
        ''' Gets the group roles.
        ''' </summary>
        ''' <returns>DataTable.</returns>
        Function GroupRoles() As DataTable

        ''' <summary>
        ''' Gets the groups by security entity.
        ''' </summary>
        ''' <returns>DataTable.</returns>
        Function GroupsBySecurityEntity() As DataTable

        ''' <summary>
        ''' Gets or sets the group roles profile.
        ''' </summary>
        ''' <value>The group roles profile.</value>
        Property GroupRolesProfile() As MGroupRoles

        ''' <summary>
        ''' Gets the profile data.
        ''' </summary>
        ''' <returns>DataRow.</returns>
        Function ProfileData() As DataRow

        ''' <summary>
        ''' Gets or sets the profile.
        ''' </summary>
        ''' <value>The profile.</value>
        Property Profile() As MGroupProfile

        ''' <summary>
        ''' Saves this instance.
        ''' </summary>
        Sub Save()

        ''' <summary>
        ''' Updates the group roles.
        ''' </summary>
        Sub UpdateGroupRoles()
    End Interface
End Namespace

